distributed/tests/test_client.py

```diff
@@ -165,6 +165,46 @@ def test_Future_exception_sync(loop):
             assert x.exception() is None
 
 
+@gen_cluster(client=True)
+def test_Future_release(c, s, a, b):
+    # Released Futures should be removed timely from the Client
+    x = c.submit(div, 1, 1)
+    yield x
+    x.release()
+    yield gen.moment
+    assert not c.futures
```

Comment:
Does tornado definitely go through all callbacks before coming back to this one?

Reply:
Not sure I understand the question... which callbacks are you talking about?

Reply:
In `x.release()` we add a callback to `_dec_ref`, then we wait a `gen.moment`, then we assume that that callback has run.  Is this a safe assumption?

Reply:
Yes, `gen.moment` should run all ready callbacks.